hledger-lib/Hledger/Data/Amount.hs

```diff
@@ -712,6 +734,9 @@ showMixedAmountDebug m | m == missingmixedamt = "(missing)"
                        | otherwise       = printf "Mixed [%s]" as
     where as = intercalate "\n       " $ map showAmountDebug $ amounts m
 
+showMixedAmountB :: AmountDisplayOpts -> MixedAmount -> WideBuilder
```

Comment:
Let's give this a haddock, and update the other two haddocks. They should help explain when to use each of these, ie mention the different normalisation behaviour.

Reply:
Haddocks are added in the next commit. Sorry, I structured this and the following three commits to minimise the diffs. I was trying to track down a subtle bug here, and couldn't figure out where it was.

I could add a note on the different normalisation behaviour, but my hope is that such a note would become redundant after this and the next PR. The eventual goal is that `MixedAmount` is always normalised by construction, and in the instances where you don't want it normalised (i.e. just the contents of `pamount`) you just use `[Amount]`. Let me know if you want me to update those haddocks now.

Reply:
Yes since it's a small thing let's add and update the haddocks now, making this commit/PR more self-contained and not dependent on future work. Haddocks don't have to be fancy, one clear sentence is usually enough.